config/src/config/consensus_config.rs

```diff
@@ -65,6 +65,40 @@ pub struct ConsensusConfig {
     // must match one of the CHAIN_HEALTH_WINDOW_SIZES values.
     pub window_for_chain_health: usize,
     pub chain_health_backoff: Vec<ChainHealthBackoffValues>,
+    pub qc_aggregator_type: QcAggregatorType,
+}
+
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub enum QcAggregatorType {
+    #[default]
+    NoDelay,
+    // TODO: Enable the delayed aggregation by default once we have tested it more.
+    // QcAggregatorType::Delayed(DelayedQcAggregatorConfig {
+    //     max_delay_after_round_start_ms: 700,
+    //     aggregated_voting_power_pct_to_wait: 90,
+    //     pct_delay_after_qc_aggregated: 30,
+    // })
+    Delayed(DelayedQcAggregatorConfig),
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct DelayedQcAggregatorConfig {
+    // Maximum Delay for a QC to be aggregated after round start (in milliseconds). This assumes that
+    // we have enough voting power to form a QC. If we don't have enough voting power, we will wait
+    // until we have enough voting power to form a QC.
+    pub max_delay_after_round_start_ms: u64,
```

Comment:
nit: probably worth having a config sanitizer ensuring this is smaller than round timeout.

Reply:
Let me look into this as a follow up.